sys/usb/usbus/cdc/acm/cdc_acm.c

```diff
@@ -0,0 +1,306 @@
+    cdcacm->iface_ctrl.hdr_gen = &cdcacm->cdcacm_hdr;
+    cdcacm->iface_ctrl.handler = handler;
+    /* Configure second interface to handle data endpoint */
+    cdcacm->iface_data.class = USB_CLASS_CDC_DATA ;
+    cdcacm->iface_data.subclass = USB_CDC_SUBCLASS_NONE;
+    cdcacm->iface_data.protocol = USB_CDC_PROTOCOL_NONE;
+    cdcacm->iface_data.hdr_gen = NULL;
+    cdcacm->iface_data.handler = handler;
+
+    /* Create required endpoints */
+    usbus_endpoint_t *ep = usbus_add_endpoint(usbus, &cdcacm->iface_ctrl,
+                                              USB_EP_TYPE_INTERRUPT,
+                                              USB_EP_DIR_IN, 8);
+    ep->interval = 255;
+    usbus_enable_endpoint(ep);
+    ep = usbus_add_endpoint(usbus, &cdcacm->iface_data, USB_EP_TYPE_BULK, USB_EP_DIR_IN, 64);
+    ep->interval = 0;
+    usbus_enable_endpoint(ep);
+    ep = usbus_add_endpoint(usbus, &cdcacm->iface_data, USB_EP_TYPE_BULK, USB_EP_DIR_OUT, 64);
+    ep->interval = 0;
+    usbus_enable_endpoint(ep);
+    usbdev_ep_ready(ep->ep, 0);
+
+    /* Add interfaces to the stack */
+    usbus_add_interface(usbus, &cdcacm->iface_ctrl);
+    usbus_add_interface(usbus, &cdcacm->iface_data);
+
+    usbus_handler_set_flag(handler, USBUS_HANDLER_FLAG_RESET);
+}
+
+static int _setup_handler(usbus_t *usbus, usbus_handler_t *handler,
+                          usbus_setuprq_state_t state, usb_setup_t *setup)
+{
+    (void)state;
+    (void)usbus;
+    usbus_cdcacm_device_t *cdcacm = (usbus_cdcacm_device_t*)handler;
+    DEBUG("Request:0x%x\n", setup->request);
+    switch(setup->request) {
+        case USB_SETUP_REQ_SET_LINE_CODING:
+            DEBUG("Value:0x%x, interface:%d, len:%d\n", setup->value,
+                                                        setup->index,
+                                                        setup->length);
+            break;
+        case USB_SETUP_REQ_SET_CONTROL_LINE_STATE:
+            if (setup->value & USB_CDC_ACM_CONTROL_LINE_DTE) {
+                DEBUG("CDC ACM: DTE Enabled\n");
+                cdcacm->state = USBUS_CDCACM_LINE_STATE_DTE;
+                _cdc_acm_flush(cdcacm);
+            }
+            else {
+                cdcacm->state = USBUS_CDCACM_LINE_STATE_DISCONNECTED;
+                DEBUG("CDC ACM: DTE disabled\n");
+            }
+            DEBUG("Value:0x%x, interface:%d, nb:%d\n", setup->value,
+                                                       setup->index,
+                                                       setup->length);
+            break;
+        default:
+            DEBUG("default handle setup rqt:0x%x\n", setup->request);
+            return -1;
+    }
+    return 0;
+}
+
+static void _handle_in(usbus_t *usbus, usbus_cdcacm_device_t *handler,
+                       usbdev_ep_t *ep)
+{
+    usbus_cdcacm_device_t *cdcacm = (usbus_cdcacm_device_t*)handler;
+    if (usbus->state != USBUS_STATE_CONFIGURED ||
+            cdcacm->state != USBUS_CDCACM_LINE_STATE_DTE) {
```

Comment:
Weird indentation and parenthesis missing.

Reply:
Fixed